cms/apps/media/forms.py

```diff
@@ -0,0 +1,47 @@
+import base64
+
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django import forms
+from io import BytesIO
+from PIL import Image
+
+from cms.apps.media.models import File
+
+
+class ImageChangeForm(forms.ModelForm):
+    changed_image = forms.CharField(
+        widget=forms.HiddenInput,
+        required=False,
+    )
+
+    class Meta:
+        model = File
+        fields = ['changed_image', 'title', 'file', 'attribution', 'copyright', 'alt_text', 'labels']
+
+    def save(self, commit=True):
+        if self.cleaned_data['changed_image']:
+            # get image data from canvas output and convert it into an image.
+            original = File.objects.filter(file=self.cleaned_data['file']).first()
+            image_data = self.cleaned_data['changed_image']
+            format, imgstr = image_data.split(';base64,')
+            original_file_name = original.file.name.split('/')[-1].split('.')[0].rsplit('_', 1)[0]
+            file_name = original_file_name + '.' + format.split('/')[-1]
```

Comment:
I am semi-sure that all this stuff around file names is unnecessary. Just save the image to `original.file.name` in your `self.cleaned_data['file'].save(....)` - if I'm not mistaken Django will _not_ overwrite the original file. (Yeah, you'll end up with `_xxxxx_xxxx_xxxx`, for values of x being some random letter, with each edit; that trade-off does not upset me if it simplifies this code.)

Reply:
...a bonus there being that it doesn't assume an underscore, or assume any kind of significance to one - think the not-uncommon case of `DSC_0001.JPG`.

Reply:
`file.name` returns `media/uploads/abc.xyz` but when I'm saving I'll want to save as just `abc.xyz` as then django will hadle the location to save to. So I'll still need the split on `/` to get rid of the location and the split on `.` to remove the filename

Reply:
since currently it converts png to png, jpg/jpeg to jpg and everything else to png.

Reply:
OK, I follow on that part. However, you're reinventing `os.path.splitext` and `os.path.basename` here - that would be a cleaner way to do the same thing.